### PR TITLE
Feature/IO-421 base architecture

### PR DESCRIPTION
**Pull request by elizaveta-seleznyova-idf**

### CODE REVIEW

File: packages/presentation/lib/navigation/base_page.dart [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import 'package:flutter/material.dart';
+import 'package:presentation/navigation/base_arguments.dart';
+
+class BasePage<T extends BaseArguments> extends Page {
+  const BasePage({
+    @required required LocalKey key,
```
Comment (vladimir-hodaryonok-idf) [inline packages/presentation/lib/navigation/base_page.dart]: As of Dart 2.12, the required keyword replaces the @required, so I think this annotation is no need here

**Review by vladimir-hodaryonok-idf — APPROVED** [review]
additionally, u can create base useCases and fix some linter warning

File: lib/di/app_injector.dart [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import 'package:data/di/data_injector.dart';
```
Comment (egor-slatvinskiy-idf) [inline lib/di/app_injector.dart:1]: due to the fact that you made a modular architecture in the master branch, I cannot leave comments on some points of this "PR", namely:
- pubspec.yaml: remove domain;
- pubspec.yaml(data): remove get_it;

**Review by egor-slatvinskiy-idf — APPROVED** [review]

File: packages/presentation/lib/navigation/base_page.dart (hunk 1 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/navigation/base_page.dart]: removed

File: lib/di/app_injector.dart (hunk 2 above)
Comment (elizaveta-seleznyova-idf) [reply lib/di/app_injector.dart:1]: removed

**Review by alexei-kruk-idf — APPROVED** [review]

File: packages/presentation/lib/app/app_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+  void _popUntil(BasePage page) {
+    final start = _appData.pages.indexWhere((e) => e.name == page.name) + 1;
+    final end = _appData.pages.length;
+    _appData.pages.removeRange(start, end);
+    _updateData();
+  }
+
+  BasePage? _currentPage() =>
+      _appData.pages.isNotEmpty ? _appData.pages.last : null;
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/app/app_bloc.dart]: lastOrNull

File: packages/presentation/lib/base/bloc_data.dart [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  );
+
+  factory BlocData.init() => BlocData<D>(
+        false,
+        null,
+      );
+
+  BlocData<D?> copy<D>() => BlocData<D?>(
+        isLoading,
+        data as D,
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/base/bloc_data.dart]: why do u need this cast?

File: packages/presentation/lib/base/bloc_data.dart [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      );
+
+  void updateParams(
+    bool? isLoading,
+    dynamic data,
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/base/bloc_data.dart]: dynamic -> D

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

File: packages/presentation/lib/base/bloc_data.dart (hunk 5 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/base/bloc_data.dart]: replaced

File: packages/presentation/lib/base/bloc_data.dart (hunk 4 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/base/bloc_data.dart]: resolved

File: packages/presentation/lib/app/app_bloc.dart (hunk 3 above)
Comment (elizaveta-seleznyova-idf) [reply packages/presentation/lib/app/app_bloc.dart]: corrected

File: packages/presentation/lib/base/bloc.dart [hunk 6]
```diff
@@ -2,32 +2,33 @@ import 'dart:async';
...
+  Stream<BlocData> get dataStream;
 
   void initState();
 
   void initArgs(T args);
 }
 
 abstract class BlocImpl<T extends BaseArguments, D> implements Bloc<T, D> {
-  final _data = StreamController<BlocData<D?>>();
+  final _data = StreamController<BlocData>();
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/base/bloc.dart]: u should keep generic. previous variant was correct

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

**Pull request merged**